### PR TITLE
Skip object validation on individual bytes in a byte array. Also, sets source compatibility to java 8.

### PR DESCRIPTION
**Pull request by d-w-johnson**

### CODE REVIEW

File: BlueDBOnDisk/src/main/java/org/bluedb/disk/serialization/validation/ObjectValidation.java [hunk 1]
```diff
@@ -144,4 +145,11 @@ private static boolean isNotEmptyCollection(Object obj) {
 		}
 		return true;
 	}
+
+	private static boolean isByteArray(Field field) {
+		if(field.getType().equals(byte[].class) || field.getType().equals(Byte[].class)) {
```
Comment (TheSirwaffle) [inline BlueDBOnDisk/src/main/java/org/bluedb/disk/serialization/validation/ObjectValidation.java:150]: Super small thing but you could just return the result of this conditional. If you think it looks more readable without doing that then you don't need to change it.

Comment (NathanGift) [inline BlueDBOnDisk/src/main/java/org/bluedb/disk/serialization/validation/ObjectValidation.java:150]: I'm fine with leaving it if you'd like, but could this method be simplified to:
 `return field.getType().equals(byte[].class) || field.getType().equals(Byte[].class);`

Comment (d-w-johnson) [reply BlueDBOnDisk/src/main/java/org/bluedb/disk/serialization/validation/ObjectValidation.java]: fixed

Comment (d-w-johnson) [reply BlueDBOnDisk/src/main/java/org/bluedb/disk/serialization/validation/ObjectValidation.java]: fixed

**Review by NathanGift — APPROVED** [review]

**Pull request merged**